### ISSUE TITLE
nvim_win_get_var threw an exception

### ISSUE THREAD

**Issue by isovector** [neovimhaskell/nvim-hs#94]
Calling `nvim_win_get_var` before my plugin is finished loading results in an exception:

```
[Neovim.Main : ERROR] Error initializing plugins: "Function threw an exception even though it was declared not to throw one: F \"nvim_win_get_var\""
[Neovim.Main : ERROR] "Function threw an exception even though it was declared not to throw one: F \"nvim_win_get_var\""
```

code:

```haskell
cornelis :: Neovim () NeovimPlugin
cornelis = do
  closeInfoWindows
  wrapPlugin $ Plugin
    { environment = env
    , exports =
        [ $(command "CornelisLoad" 'load) [CmdSync Async]
        , $(command "CornelisSolve" 'solveOne) [CmdSync Async]
        , $(command "CornelisMakeCase" 'caseSplit) [CmdSync Async]
        ]
    }


cornelisWindowVar :: String
cornelisWindowVar = "cornelis_window"


closeInfoWindows :: Neovim env ()
closeInfoWindows = do
  ws <- vim_get_windows
  for_ ws $ \w ->
    window_is_valid w >>= \case
      False -> pure ()
      True -> do
        nvim_win_get_var w cornelisWindowVar >>= \case
          ObjectBool True -> do
            nvim_win_close w True
          _ -> pure ()
```

Is this a bug in nvim-hs, or in nvim? Is there a good workaround here?

**Comment by isovector** [comment]
Looks like running `closeInfoWindows` asynchronously after a `threadDelay` works around the issue, but it'd be nice to have a better solution here.

**Comment by saep** [comment]
The error message is terrible. I have pushed a commit to master that also prints original exception. I have also created a pull request #95 which removes the suppression of the exception entirely. Maybe you want to provide some feedback or test it out? ;-)

The Neovim RPC API has fields which state whether a method can fail (i.e. throw exceptions). Since every method can fail, I decided to refactor the generated API to silently throw exceptions instead of always returning `Either NeovimException result`. Stupidly, I used the suppression code for methods that should't fail to impelent this instead of simply using `throwIO`. I've even written in the changelog for version 2.0.0.0 that `catch` should be used if you want to handle exceptions, but the change has made it really awkward to do.

As to the workaround with using `threadDelay`: Maybe we can simply add a field `runAfterPluginLoads` to `Plugin` with type `Neovim () ()`. To preserve backwards-compatibility, we could also add another constructor which has the added field. Or maybe break backwards-compatiblity a little and create a function that createss a `Plugin` with the 2 mandatory fields as arguments.
